src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -563,12 +563,18 @@ protected override void ProcessRecord()
 
                             using HttpResponseMessage response = GetResponse(client, request, handleRedirect);
 
+                            if (TimeoutSec > 0)
```

Comment:
This use of TimeOut sec is not correct, it should indicate the timeout before the connetion to the server, not during the download
test:
```ps1
iwr https://software-static.download.prss.microsoft.com/dbazure/988969d5-f34g-4e03-ac9d-1f9786c66751/22621.525.220925-0207.ni_release_svc_refresh_CLIENTENTERPRISEEVAL_OEMRET_x64FRE_en-us.iso -Timeout 5
#--> fails after 5 seconds
```

Reply:
Yes, I did ask about this in #12249 comments. I thought this needed some discussion but wanted to get a working prototype out for discussion.

Should we use a second parameter like ReceiveTimeout with default of 5 minutes or do something else?

Reply:
I think the desired behaviour is checking if the download is stuck for 5 minutes and then closing the download with a warning message. At the moment your implementation halts the successful download after a timespan (-ReceiveTimeout) has passed. This could be a new feature (you should open an issue to discuss if we should implement it), but it's not the correct solution for issue #12249.
This PR at the moment fixes #16145 (Ctrl-c on download stuck because of no internet)

Reply:
Ok, I see. I think implementing what you describe is likely to be a more complex patch as it would need to monitor the stream progress and adjust cancellation timeouts whenever data is received. Given this PR still address #16145 I think it would be good to submit (with the timeout removed) so that issue can be closed, if it is agreed that this is the correct way to deal with that issue.

Reply:
We will have to wait for @iSazonov 's rewiew, but it looks good to me 👍

Reply:
My comment https://github.com/PowerShell/PowerShell/issues/12249#issuecomment-1466026974

Reply:
@iSazonov I have responded to your comment in the issue. I don't think using a separate task is the right solution as I have explained there. Could you please review this PR as it demonstrates that using a task is not necessary to support cancellation. Provided you are okay with the PR I will go back and fix the codefactor issues in the code base that are being flagged.

I will try to build timeout cancellation on top of this PR once it is approved using the approach I outlined in #12249.

Reply:
FYI, I have fixed reported codefactor issues and merged recent changes from master branch.